### DISCUSSION TITLE
install.ztnet.network shell script

### DISCUSSION THREAD

**Discussion by tinola** [sinamics/ztnet#184] [Show and tell]
Hi @sinamics 

I've made small improvements in your bash script :) I think they'll be useful in the future.

Patch format as follows:

```
--- install_orig.sh	2023-10-27 17:14:21.714970671 +0200
+++ install.sh	2023-10-27 17:16:57.896384798 +0200
@@ -46,7 +46,7 @@ fi
 printf "\n\n${YELLOW}Welcome to the installation script.${NC}\n"
 printf "${YELLOW}This script will perform the following actions:${NC}\n"
 printf "  1. Check if PostgreSQL is installed. If not, it will be installed.\n"
-printf "  2. Check if Node.js version 18 is installed. If not, it will be installed.\n"
+printf "  2. Check if Node.js version "$NODE_MAJOR" is installed. If not, it will be installed.\n"
 printf "  3. Clone ztnet repo into /tmp folder and build artifacts .\n"
 printf "  4. Copy artifacts to /opt/ztnet folder.\n"
 printf "${YELLOW}Please note:${NC}\n"
@@ -95,12 +95,12 @@ if ! command_exists psql; then
     fi
 fi
 
-# Install Node.js if it's not installed or if installed version is not 18
+# Install Node.js if it's not installed or if installed version is not the number defined in 'NODE_MAJOR' variable
 if ! command_exists node; then
   INSTALL_NODE=true
 else
   NODE_VERSION=$(node -v | cut -d 'v' -f 2 | cut -d '.' -f 1)
-  if [ "$NODE_VERSION" -lt 18 ]; then
+  if [ $NODE_VERSION -lt $NODE_MAJOR ]; then
     INSTALL_NODE=true
   fi
 fi

```

Download:
[install.sh.patch](https://github.com/sinamics/ztnet/files/13191017/install.sh.patch)

Cheers!

**Comment by sinamics** [comment]
awesome, much appreciated. 
I need all help i can get. 


Thx

**Comment by tinola** [comment]
Well... There're another issues I've just discovered with this script...

I've installed clean Debian 12 without X and other stuff, and when I tried to install using `curl -s http://install.ztnet.network | sudo bash` 
that fails because of no `curl` package is installed.

So I installed `curl`, then again I ran `curl -s http://install.ztnet.network | sudo bash`
That again fails because `sudo` was not found ;)
Yes! Clean Debian 12 install in text mode does not have `sudo` installed by default!

Thoughts... In the documentation we should mention at the beginning, that `curl` and `sudo` packages are mandatory to install `ztnet` via `"curl -s http://install.ztnet.network | sudo bash"` command.

I don't know if this applies to Ubuntu... Didn't try.

Regards!

**EDIT (clean install of Debian 12):**
```
root@debian-hv:~# curl -s http://install.ztnet.network | sudo bash
bash: sudo: command not found
bash: curl: command not found
```

**Comment by sinamics** [comment]
Thx.
I have updated the readme and added your patch.
I also added the install.ztnet code into this main repo for better visibility.
https://github.com/sinamics/ztnet/tree/main/install.ztnet

**Comment by sinamics** [comment]
https://ztnet.network/installation/linux

**Comment by tinola** [comment]
No no!

We must bear in mind, that clean Debian install does not have `sudo` installed by default as well :/

According to:
![manual](https://github.com/sinamics/ztnet/assets/3160125/cd34b222-7cb0-4708-9b1a-5cee2c937099)

I've removed "sudo" form command:
```
root@debian-hv:~# curl -s http://install.ztnet.network | bash


Welcome to the installation script.
This script will perform the following actions:
  1. Check if PostgreSQL is installed. If not, it will be installed.
  2. Check if Node.js version 18 is installed. If not, it will be installed.
  3. Clone ztnet repo into /tmp folder and build artifacts .
  4. Copy artifacts to /opt/ztnet folder.
Please note:
  - You will have the option to set a custom password for the PostgreSQL user 'postgres'.
Press space to proceed with the installation...bash: line 70: sudo: command not found
```
Agrrr, it fails ;)

So I propose in the manual at the top of topic to change:

```
Install curl if it is not already installed:

sudo apt update && sudo apt install curl
```

to:

```
Login as **root** on your system, then install 'curl' and 'sudo' if it is not already installed:

apt update && apt install -y sudo curl

```
